src/dsp_tools/commands/xmlupload/models/ontology_diagnose_models.py

```diff
@@ -0,0 +1,102 @@
+# sourcery skip: use-fstring-for-concatenation
+
+from dataclasses import dataclass, field
+from pathlib import Path
+
+import pandas as pd
+
+from dsp_tools.models.exceptions import UserError
+
+
+@dataclass(frozen=True)
+class Ontology:
+    """This class saves the properties and the classes from an ontology."""
+
+    classes: list[str] = field(default_factory=list)
+    properties: list[str] = field(default_factory=list)
+
+
+@dataclass
+class OntoCheckInformation:
+    """This class saves information that is needed to check if consistency with the ontology."""
+
+    default_ontology_prefix: str
+    onto_lookup: dict[str, Ontology]
+    save_location: Path
+
+
+@dataclass(frozen=True)
+class InvalidOntologyElements:
+    """This class saves and prints out the information regarding ontology classes and properties
+    that are in the XML but not the ontology."""
+
+    save_path: Path
+    classes: list[list[str]] = field(default_factory=list)
+    properties: list[list[str]] = field(default_factory=list)
+
+    def execute_problem_protocol(self) -> None:
+        """
+        This function is executed if there are any elements in properties or classes.
+        If there are more than 100 entries combined, then the result is also saved as an excel.
+
+        Raises:
+            UserError: If properties or classes have any entries.
+        """
+        extra_separator = "\n\n----------------------------\n\n"
+        msg = "Some property and or class type(s) used in the XML are unknown:" + extra_separator
+        cls_msg = self._print_problem_string_cls()
+        if cls_msg:
+            msg += cls_msg + extra_separator
+        prop_msg = self._print_problem_string_props()
+        if prop_msg:
+            msg += prop_msg
+        if len(self.classes) + len(self.properties) > 50:
+            ex_name = "InvalidOntologyElements_in_XML.xlsx"
```

Comment:
Not so user friendly :-) What about 

```suggestion
            ex_name = f"{orig_filename}_syntax_errors.xlsx"
```